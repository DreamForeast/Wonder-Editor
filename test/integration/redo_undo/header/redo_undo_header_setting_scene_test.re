open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: header->setting->scene", () => {
    let sandbox = getSandboxDefaultVal();

    let _simulateSetSkyboxCubemapTexture = () => {
      let assetTreeData =
        MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

      HeaderSettingTool.Scene.Skybox.setCubemapTextureToSceneSkybox(
        MainEditorAssetCubemapNodeTool.getCubemapTextureComponent(
          ~nodeId=
            MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
              assetTreeData,
            ),
          (),
        ),
      );
    };

    let _beforeEach = () => MainEditorSceneTool.prepareScene(sandbox);

    let _afterEach = () => ();

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test set skybox->cubemap texture", () =>
      RedoUndoTool.testRedoUndoOneStep(
        sandbox,
        "test redo/undo one step",
        (_simulateSetSkyboxCubemapTexture, _beforeEach, _afterEach),
        HeaderSettingTool.UI.buildSettingSceneModal,
      )
    );
  });