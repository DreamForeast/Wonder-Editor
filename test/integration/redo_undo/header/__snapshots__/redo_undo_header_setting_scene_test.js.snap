// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: header->setting->scene test set skybox->cubemap texture test redo/undo one step test redo operate test redo one step undo step which from first to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-settingScene-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      SettingSceneModal
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <div
        className="setting-scene-ambient"
      >
        <div
          className="header-item"
        >
          <div
            className="component-item"
          >
            <article
              className="inspector-item"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#000000",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
      </div>
      <div
        className="setting-scene-skybox"
      >
        <div
          className="item-header"
          title="set scene skybox->cubemap"
        >
          Skybox Cubemap
        </div>
        <div
          className="item-content"
        >
          <div
            className="cubemap-img"
            onClick={[Function]}
          >
            <img
              src="./public/img/cubemap.png"
            />
            <span>
              cubemap1
            </span>
          </div>
          <button
            className="cubemap-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`redo_undo: header->setting->scene test set skybox->cubemap texture test redo/undo one step test undo operate test not undo 1`] = `
<article
  className="wonder-settingScene-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      SettingSceneModal
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <div
        className="setting-scene-ambient"
      >
        <div
          className="header-item"
        >
          <div
            className="component-item"
          >
            <article
              className="inspector-item"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#000000",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
      </div>
      <div
        className="setting-scene-skybox"
      >
        <div
          className="item-header"
          title="set scene skybox->cubemap"
        >
          Skybox Cubemap
        </div>
        <div
          className="item-content"
        >
          <div
            className="cubemap-img"
            onClick={[Function]}
          >
            <img
              src="./public/img/cubemap.png"
            />
            <span>
              cubemap1
            </span>
          </div>
          <button
            className="cubemap-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`redo_undo: header->setting->scene test set skybox->cubemap texture test redo/undo one step test undo operate test undo one step undo step which from first to zero 1`] = `
<article
  className="wonder-settingScene-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      SettingSceneModal
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <div
        className="setting-scene-ambient"
      >
        <div
          className="header-item"
        >
          <div
            className="component-item"
          >
            <article
              className="inspector-item"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#000000",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
      </div>
      <div
        className="setting-scene-skybox"
      >
        <div
          className="item-header"
          title="set scene skybox->cubemap"
        >
          Skybox Cubemap
        </div>
        <div
          className="item-content"
        >
          <div
            className="cubemap-img"
            onClick={[Function]}
          />
          <button
            className="cubemap-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </div>
    </div>
  </div>
</article>
`;
